src/bp/bpfs.ts

```diff
@@ -0,0 +1,229 @@
+
+      if (_.isEmpty(blockingChanges) || useForce) {
+        this._printChangeList(changeList)
+
+        if (dryRun) {
+          console.log(chalk.yellow(`Dry run completed. Nothing was pushed to server`))
+          return process.exit()
+        }
+
+        console.log(chalk.blue(`Pushing local changes to ${this.serverUrl}... ${useForce ? '(using --force)' : ''}`))
+
+        await axiosClient.post('update', archive)
+        console.log(chalk.green(`Successfully pushed ${localFiles.length} local files to remote server!`))
+      } else {
+        this._printOutOfSync()
+        this._printChangeList(changeList)
+        console.log(chalk.red(`Nothing was pushed on the remote server.`))
+      }
+    } catch (err) {
+      const error = err.response ? `${err.response.statusText} (${err.response.status})` : err.message
+      this._endWithError(`Could not push changes: ${error}`)
+    }
+  }
+
+  private async _filesCount(directory: string) {
+    const files: string[] = await Promise.fromCallback(cb =>
+      glob('**/*', { cwd: directory, nodir: true, dot: true }, cb)
+    )
+    return files.length
+  }
+
+  private _processChanges(data: FileChanges) {
+    const changeList = _.flatten(data.map(x => x.changes))
+    return {
+      localFiles: _.flatten(data.map(x => x.localFiles)),
+      blockingChanges: changeList.filter(x => blockingActions.includes(x.action)),
+      changeList
+    }
+  }
+
+  private _getPushAxiosClient(archiveSize: number): AxiosInstance {
+    return axios.create({
+      baseURL: `${this.serverUrl}/api/v1/admin/versioning`,
+      headers: {
+        Authorization: `Bearer ${this.authToken}`,
+        'Content-Type': 'application/tar+gzip',
+        'Content-Disposition': `attachment; filename=archive_${Date.now()}.tgz`,
+        'Content-Length': archiveSize
+      }
+    })
+  }
+
+  private _getPullAxiosClient(): AxiosInstance {
+    return axios.create({
+      baseURL: `${this.serverUrl}/api/v1/admin/versioning`,
+      headers: {
+        Authorization: `Bearer ${this.authToken}`
+      },
+      responseType: 'arraybuffer'
+    })
+  }
+
+  private async _clearDir(destination: string): Promise<void> {
+    if (fse.existsSync(destination)) {
+      return Promise.fromCallback(cb => rimraf(destination, cb))
+    }
+  }
+
+  private _printLine({ action, path, add, del }): string {
+    if (action === 'add') {
+      return chalk.green(` + ${path}`)
+    } else if (action === 'del') {
+      return chalk.red(` - ${path}`)
+    } else if (action === 'edit') {
+      return ` o ${path} (${chalk.green('+' + add)} / -${chalk.redBright(del)})`
+    }
+    return ''
+  }
+
+  private _printOutOfSync() {
+    console.log(chalk`
+{bold Out of sync! }
+  You have changes on your file system that aren't synchronized to the remote environment.
+
+  (Replace {bold "push"} with {bold "pull"} in your command to pull remote changes on your file system)
```

Comment:
I think we should assume that if the user uses push command, it knows that pull exist. I think the following would be better:

```
chalk.yellow(Conflict warning) Remote has changes that are not synced to your environment. Backup your changes and use "pull" to get those changes on your file system.

Use --force to overwrite remote changes with yours.
```